src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,402 @@
+                temp += "";
+            }
+            temp += "\n";
+            fw.write(temp);
+        }
+        fw.close();
+    }
+
+    public static void wrapWriteToDisk(ArrayList<Task> tasks){
+        try{
+            writeToDisk(tasks);
+        } catch (IOException e) {
+            System.out.println("Issues with writing to .txt file");
+        }
+    }
+
+    public static void loadDataFromDisk(ArrayList<Task> tasks) throws FileNotFoundException {
+        File dataFile = new File(String.valueOf(dataFileDirectory));
+        Scanner s = new Scanner(dataFile);
+
+        while(s.hasNext()){
+            //format I am using is, to use index 0 to store the status.
+            //E.g:
+            //1|event|memes   r|/at|   next Friday
+            String temp = s.nextLine();
+            String[] command = temp.split("\\|");
+            Task tempTask = populateArrayList(Arrays.copyOfRange(command,1, command.length), tasks);
+            if (command[0].equals("1")){
+                tempTask.setAsDone();
+            }
+        }
+        s.close();
+    }
+
+    public static void createNewFile() {
+        try{
+            // If directory does not exist, then create the directory
+            // If directory exists already, just does nothing
+            checkForDirectory();
+        } catch (IOException e) {
+            System.out.println("A serious error has occurred");
+        }
+
+        try{
+            // If file does not exist, then create the file
+            checkForFile();
+        } catch (IOException e) {
+            System.out.println("A serious error has occurred");
+        }
+    }
+
+    public static void checkForFile() throws IOException {
+        if(!Files.exists(dataFileDirectory)){
+            File newFile = new File(String.valueOf(dataFileDirectory));
+            newFile.createNewFile();
+        }
+    }
+
+    public static void checkForDirectory() throws IOException {
+        if(!Files.exists(dataDirectory)){
+            Files.createDirectory(dataDirectory);
+        }
+    }
+
+    public static void runProgram() {
+        Scanner in = new Scanner(System.in);
+        String userInput;
+
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        // Check if there is such a file. Otherwise, create a new file.
+        try{
+            loadDataFromDisk(tasks);
+        } catch (FileNotFoundException e){
+            createNewFile();
+        }
+
+        do {
+            userInput = in.nextLine();
+
+            String[] command = userInput.split(" ");
+
+            if(isSpecialCharacterPresent(userInput)){
+                continue;
+            }
+
+            Task newItem = populateArrayList(command, tasks);
+            if(newItem != null){
+                newItemMessage(tasks, newItem);
+                wrapWriteToDisk(tasks);
+            }
+        } while (!userInput.equals("bye"));
+    }
+
+    public static Task populateArrayList(String[] command, ArrayList<Task> tasks){
```

Comment:
The name of this method is a bit misleading. Which ArrayList? Does this method really "populate" the ArrayList "tasks"? From my perspective, this method actually takes different commands and executes them based on their types.